### PR TITLE
Featureimage/release 2.3.1

### PR DESCRIPTION
**Pull request by stark-dev**
Backport from 2.3.1 release
- new group/feature Virtualization Settings
- Prevent restore when license is expired

### CODE REVIEW

**Review by jimklimov — APPROVED** [review]
LGTM for currently proposed backports :) But safer to merge after checking that the parts play together well in the other image (and also merge this consistently with other backports).

**Comment by stark-dev** [comment]
To be merged after `etn-licensing` PR #184

File: server/src/fty_srr_worker.cc [hunk 1]
```diff
@@ -427,6 +432,60 @@ namespace srr
         return response;
     }
 
+    static std::string zmsg_popstring(zmsg_t *resp){
```
Comment (aquette) [inline server/src/fty_srr_worker.cc:435]: we should really put that into common_mlm, since it's present elsewhere (*-rest esp.)

**Review by aquette — APPROVED** [review]
LGTM

**Pull request closed without merging**